File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        globalMax = nums[0]
        sofarMax = nums[0]
        
        for i in range(1, len(nums)):
            sofarMax = max(sofarMax + nums[i], nums[i])
            globalMax = max (globalMax, sofarMax)
        
        return globalMax

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int globalMax = nums[0];
        int prevAns   = nums[0];
        
        for (int i = 1; i < nums.size(); i++) {
            prevAns = max(prevAns + nums[i], nums[i]);
            globalMax = max(globalMax, prevAns);
        }
        
        return globalMax;
    }
};